### PR TITLE
Add Shadow Candle and Brain of Confusion to `PlayerAddBuffWhitelist`, improve rejection messages

### LINKED ISSUE #2799: Brain of Confusion Dodge Removes All Buffs and Debuffs
* TShock version: TShock 5
* TShock build number (if known):

#### Reproduction steps (if applicable)?
1. Equip Brain of Confusion
2. Chug some potions
3. Activate Brain of Confusion's dodge
4. ALL buff duration countdowns turn to 0s.
Console said: 
"GetDataHandlers: VERBOSE: GetDataHandlers HandlePlayerBuffList handled event and sent data tchravis"
when the Brain of Confusion activated and wiped all of my buffs.

#### Any stack traces or error messages (if known)?
`Unknown`
PUT SUPER LONG ERROR MESSAGES IN THE TICK MARKS
```
#### Any screenshots?

It seems like the Player Buff Handler is doing something wrong, the red circle is what was logged when the Brain of Confusion's dodge activated. 
Even the Potion Sickness Debuff was set to 0 seconds, and 'permanent' buffs like Happy also briefly hit 0s. 
![image](https://user-images.githubusercontent.com/116663950/200007325-4abda3ac-7792-44a8-91f0-383558a0bc44.png)

#### Any log messages from files that end in `.log` or `.txt`? What are the last 100 log messages from the server console?
[2022-11-04_09-52-42.log](https://github.com/Pryaxis/TShock/files/9939544/2022-11-04_09-52-42.log)

#### What plugins and what versions of those plugins are you running?
Crossplay, Chestroom, FindPermission, Freeze, House Regions, NPC Blocker, Region View, TCR Discord, Terraria Chat Relay, WorldRefill
Though I doubt that any of these plugins are affecting player buffs.

If I didn't provide any logs this issue, please close my issue immediately. I'm sorry for the inconvenience.

### PR DESCRIPTION
**Pull request by drunderscore**
Fixes #2799
Fixes [issue in Discord](https://discord.com/channels/479657350043664384/479657350043664386/1040411699632672858) reported by Kojiro (which was reported to them by another individual)

### CODE REVIEW

File: TShockAPI/Bouncer.cs [hunk 1]
```diff
@@ -1867,41 +1879,43 @@ internal void OnPlayerBuff(object sender, GetDataHandlers.PlayerBuffEventArgs ar
...
+					args.Player.Name, args.Player.Index, type, id, time, reason));
 				args.Handled = true;
+
+				if (shouldResync)
+					args.Player.SendData(PacketTypes.PlayerBuff, number: id);
+			}
+
+			if (id >= Main.maxPlayers)
+			{
+				Reject("Target ID is over player capacity", false);
```
Comment (sgkoishi) [inline TShockAPI/Bouncer.cs]: Would it be better to wrap all those `Reject` reasons with `GetString`?

Comment (drunderscore) [reply TShockAPI/Bouncer.cs]: Oops! Totally right, just forgot, I'll be sure to amend those :^)

**Review by sgkoishi — DISMISSED** [review]

**Review by hakusaro — CHANGES_REQUESTED** [review]
Unless @pontaoski disagrees, I think this introduces word puzzles, and therefore, shouldn't be merged as-is. In particular, the buff reason shouldn't be interpolated blindly into the message on L1884 of Bouncer.cs -- ideally you should pass the message each time.

**Comment by drunderscore** [comment]
I've updated the commit changing the log messages to not interpolate multiple sentences together. Instead, each call site now logs it's own message.

Otherwise, the rationale for that commit is the same: improve debug output to speed up triage of buff whitelist issues.

**Comment by ATFGK** [comment]
"Sandstorm" and "Brand of the Inferno" will also make buff disappear

**Pull request merged**